conda_smithy/variant_algebra.py

```diff
@@ -0,0 +1,153 @@
+import yaml
+import toolz
+from conda_build.utils import ensure_list
+import conda_build.variants as variants
+from conda.models.version import VersionOrder
+from functools import partial
+
+
+def parse_variant(variant_file_content: str, config=None):
+    """
+    Parameters
+    ----------
+    variant_file_content : str
+        The loaded vaiant contents.  This can include selectors etc.
+    """
+    if not config:
+        from conda_build.config import Config
+
+        config = Config()
+    from conda_build.metadata import select_lines, ns_cfg
+
+    contents = select_lines(
+        variant_file_content, ns_cfg(config), variants_in_place=False
+    )
+    content = yaml.load(contents, Loader=yaml.loader.BaseLoader) or {}
+    variants.trim_empty_keys(content)
+    return content
+
+
+def _version_order(v, ordering=None):
+    if ordering is not None:
+        return ordering.index(v)
+    else:
+        try:
+            return VersionOrder(v)
+        except:
+            return v
+
+
+def variant_key_add(k, v_left, v_right, ordering=None):
+    """Version summation adder.
+
+    This takes the higher version of the two things.
+    """
+    out_v = []
+    common_length = min(len(v_left), len(v_right))
+    for i in range(common_length):
+        e_l, e_r = v_left[i], v_right[i]
+        print(i, e_l, e_r)
```

Comment:
Is this print needed, or is it there for debugging?

Reply:
just debuggin